client/src/components/calendar.js

```diff
@@ -0,0 +1,147 @@
+/** @jsx jsx */
+import React from "react";
+import { jsx } from "@emotion/core";
+import dateFns from "date-fns";
+
+function Calendar({
+  start,
+  end,
+  loading,
+  onSelected,
+  onContinue,
+  getData,
+  events,
+  selected
+}) {
+  const [cursor, setCursor] = React.useState(dateFns.startOfWeek(new Date()));
+  const [dates, setDates] = React.useState([]);
+  const hours = Array.from(Array(end - start).keys()).map(i => i + start);
+
+  React.useEffect(() => {
+    const times = Array.from(Array(7).keys()).map((_, index) => {
+      return dateFns.format(dateFns.addDays(cursor, index), "YYYY-MM-D");
+    });
+    setDates(times);
+  }, [cursor]);
+
+  React.useEffect(() => {
+    if (dates.length > 0) {
+      getData(dates[0], dates[dates.length - 1]);
+    }
+  }, [dates]);
+
+  function today() {
+    setCursor(dateFns.startOfWeek(new Date()));
+  }
+
+  function next() {
+    if (dateFns.differenceInCalendarDays(cursor, new Date()) < 30) {
+      setCursor(dateFns.addDays(cursor, 7));
+    } else {
+      alert("Please select an available booking date");
+    }
+  }
+
+  function back() {
+    if (
+      dateFns.differenceInCalendarDays(
+        cursor,
+        dateFns.startOfWeek(new Date())
+      ) >= 7
+    ) {
+      setCursor(dateFns.subDays(cursor, 7));
+    } else {
+      alert("Please select an available booking date");
```

Comment:
It seems as a practical way to validate.

Reply:
Thanks, man `:)`, we will always carry you in our hearts